modules/ding_place2book/ding_place2book.module

```diff
@@ -442,61 +442,68 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
     $p2b = ding_place2book_instance();
     $event = $p2b->getEvent($event_maker_id, $event_id);
     $meta_data = $event->meta_data;
+    $event_state = $meta_data->event_state;
     $start = strtotime($event->begin_at);
 
-    if (time() < $start) {
-      // Go through the event state.
-      switch ($meta_data->event_state) {
-        case 'published':
-          // Check if waiting list is activated.
-          switch ($meta_data->waiting_list) {
-            case 'true':
-              $status = 'waiting-list';
-              break;
-
-            default:
-              // Go through the sales status.
-              switch ($meta_data->sales_status) {
-                case 'open':
-                  // Check if there is any available tickets for sale.
-                  if ($meta_data->available_tickets > 0 || strval($meta_data->available_tickets) == '∞') {
-                    $status = 'open';
-                  }
-                  else {
-                    $status = 'sold-out';
-                  }
-                  break;
-
-                default:
-                  $status = 'closed';
-                  break;
-              }
-              break;
+    // We start out with the closed status and look in the event data to prove
```

Comment:
Why define anything at all?

Reply:
So you would just use an empty string?

The idea was to start out with status closed and only check the event for the other statuses.

Reply:
It this is not what you do; you just set the status. If you only wanna check other status’ you would need to wrap everything around if!=closed, right?

Reply:
I'm not sure I follow you, but nevermind I ended up with another approach.